### PR TITLE
NOX-646 Fix Swagger error for ApiRouteMappings

### PR DESCRIPTION
**Pull request by emir-sehovic1**

### CODE REVIEW

File: src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs [hunk 1]
```diff
@@ -74,13 +78,38 @@ public override ValidationResult Validate(NoxSolution topNode, ApiConfiguration
...
+            if (!definedParameters.Any(x => string.Equals(x, parameter, StringComparison.InvariantCultureIgnoreCase)))
+            {
+                result.AddError(nameof(RequestInput), $"Endpoint [{Name}] defines a parameter [{parameter}] in the route that is not defined in the [{nameof(RequestInput)}] section.");
+            }
+        }
+
         return result;
     }
 
+    private IEnumerable<string> ExtractParametersFromRoute()
```
Comment (rochar) [inline src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs]: I would suggest to return an IReadonlyCollection, so its clear that the result is not going to be computed
```suggestion
    private IReadOnlyCollection<string> ExtractParametersFromRoute()
```

File: src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs [hunk 2]
```diff
@@ -74,13 +78,38 @@ public override ValidationResult Validate(NoxSolution topNode, ApiConfiguration
...
+    {
+        string parametersPattern = @"\{(\w+)}";
+        MatchCollection matches = Regex.Matches(Route, parametersPattern, RegexOptions.None, TimeSpan.FromMilliseconds(100));
+
+        List<string> parameterNames = new();
```
Comment (rochar) [inline src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs]: matches.Count is the expected size of the list?
```suggestion
        List<string> parameterNames = new(matches.Count);
```

File: src/Nox.Lib/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Nox.Extensions;
+
+internal static class StringExtensions
+{
+    public static string StripQueryParameters(this string input) => input.Split('?')[0];
```
Comment (rochar) [inline src/Nox.Lib/Extensions/StringExtensions.cs]: this approach will require less memory allocation?
```suggestion
    public static string StripQueryParameters(this string input) =>input.Substring(0, input.IndexOf(',?'));
```

File: src/Nox.Lib/Extensions/StringExtensions.cs
Comment (rochar) [inline src/Nox.Lib/Extensions/StringExtensions.cs:1]: Although this seems a string extension, is actually more a URI extension, I would suggest to move this method to ApiRouteMappingDocumentFilter.cs and make it private for now

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, just a couple suggestions

**Comment by emir-sehovic1** [comment]
> looks good, just a couple suggestions

@rochar Thank you for the feedback, I've applied the changes

**Review by anna-naumova25 — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**